pkg/syncer/fullsync.go

```diff
@@ -272,13 +275,67 @@ func CsiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer, vc s
 	go fullSyncDeleteVolumes(ctx, volToBeDeleted, metadataSyncer, &wg, migrationFeatureStateForFullSync, volManager, vc)
 	wg.Wait()
 
+	// Sync VolumeInfo CRs
+	if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+		volumeInfoCRFullSync(ctx, k8sPVMap, vc)
+	}
+
 	cleanupCnsMaps(k8sPVMap)
 	log.Debugf("FullSync: cnsDeletionMap at end of cycle: %v", cnsDeletionMap)
 	log.Debugf("FullSync: cnsCreationMap at end of cycle: %v", cnsCreationMap)
 	log.Infof("FullSync: end")
 	return nil
 }
 
+// volumeInfoCRFullSync creates VolumeInfo CR if it does not already exist for a volume.
+// It also deletes VolumeInfo CR if its corresponding PV does not exist.
+func volumeInfoCRFullSync(ctx context.Context, k8sPvs map[string]string, vc string) {
+	log := logger.GetLogger(ctx)
+
+	for _, volumeID := range k8sPvs {
+		crExists, err := volumeInfoService.VolumeInfoCrExistsForVolume(ctx, volumeID)
+		if err != nil {
+			log.Errorf("failed to find VolumeInfo CR for volume %s."+
+				"Error: %+v", volumeID, err)
+			continue
+		}
+
+		// Create VolumeInfo CR if not found.
+		if !crExists {
+			err := volumeInfoService.CreateVolumeInfo(ctx, volumeID, vc)
```

Comment:
How do you know that VolumeID is associated with the supplied vCenter server?

Reply:
As discussed, we are only considering Pvs from a particular VC.